Sources/GATTBodyCompositionMeasurement.swift

```diff
@@ -0,0 +1,720 @@
+             .userID,
+             .basalMetabolism,
+             .musclePercentage,
+             .muscleMass,
+             .fatFreeMass,
+             .softLeanMass,
+             .bodyWaterMass,
+             .impedance,
+             .weight,
+             .height,
+             .multiplePacket
+        ]
+    }
+    
+    public enum MassUnit: UInt16 {
+        
+        /// Mass kilogram
+        case kilogram = 0x2702
+        
+        /// Mass pound
+        case pound = 0x27B8
+        
+        public init?(unit: UnitIdentifier) {
+            
+            self.init(rawValue: unit.rawValue)
+        }
+        
+        public var unit: UnitIdentifier {
+            
+            return UnitIdentifier(rawValue: rawValue)
+        }
+    }
+    
+    public enum LengthUnit: UInt16 {
+        
+        /// Length metre
+        case metre = 0x2701
+        
+        /// length inch
+        case inch = 0x27A2
+        
+        public init?(unit: UnitIdentifier) {
+            
+            self.init(rawValue: unit.rawValue)
+        }
+        
+        public var unit: UnitIdentifier {
+            
+            return UnitIdentifier(rawValue: rawValue)
+        }
+    }
+    
+    public struct BodyMass {
+        
+        public typealias MassUnit = GATTBodyCompositionMeasurement.MassUnit
+        
+        internal static let length = MemoryLayout<UInt16>.size
+        
+        public var unit: MassUnit
+        
+        public var rawValue: UInt16
+        
+        public init?(rawValue value: UInt16, unit: MassUnit) {
+            
+            self.rawValue = value
+            self.unit = unit
+        }
+    }
+    
+    public struct BodyLength {
+        
+        public typealias LengthUnit = GATTBodyCompositionMeasurement.LengthUnit
+        
+        internal static let length = MemoryLayout<UInt16>.size
+        
+        public var unit: LengthUnit
+        
+        public var rawValue: UInt16
+        
+        public init?(rawValue value: UInt16, unit: LengthUnit) {
+            
+            self.rawValue = value
+            self.unit = unit
+        }
+    }
+}
+
+// MARK: - BluetoothUnits -
+public struct GATTBodyPercentage: BluetoothUnit {
+    
+    internal static let length = MemoryLayout<UInt16>.size
+    
+    public static var unitType: UnitIdentifier { return .percentage }
+    
+    public var rawValue: UInt16
+    
+    public init(rawValue value: UInt16) {
+        
+        self.rawValue = value
+    }
+}
+
+extension GATTBodyPercentage: Equatable {
+    
+    public static func == (lhs: GATTBodyPercentage, rhs: GATTBodyPercentage) -> Bool {
+        
+        return lhs.rawValue == rhs.rawValue
+    }
+}
+
+extension GATTBodyPercentage: CustomStringConvertible {
+    
+    public var description: String {
+        
+        return "\(rawValue)%"
+    }
+}
+
+extension GATTBodyPercentage: ExpressibleByIntegerLiteral {
+    
+    public init(integerLiteral value: UInt16) {
```

Comment:
👍🏻 Here is `value` parameter adds extra clarification and allows a nicer variable nice while satisfying the protocol, so here this is a good API design.